apache-libraries-2/src/main/java/com/baeldung/avro/SerializationDeserializationLogic.java

```diff
@@ -0,0 +1,40 @@
+package com.baeldung.avro;
+
+import generated.avro.Car;
+import org.apache.avro.file.DataFileReader;
+import org.apache.avro.file.DataFileWriter;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.DatumWriter;
+import org.apache.avro.specific.SpecificDatumReader;
+import org.apache.avro.specific.SpecificDatumWriter;
+
+import java.io.File;
+import java.io.IOException;
+
+public class SerializationDeserializationLogic {
+
+    static void serializeCar(Car car) {
+
+        DatumWriter<Car> userDatumWriter = new SpecificDatumWriter(Car.class);
+
+        try (DataFileWriter<Car> dataFileWriter = new DataFileWriter(userDatumWriter)) {
+            dataFileWriter.create(car.getSchema(), new File("cars.avro"));
+            dataFileWriter.append(car);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```

Comment:
This is not a good way of dealing with exceptions so we shouldn't teach the reader this. Please either let the exception get rethrown, or at the very least log the exception using SLF4J. Similar in deserializeCar.